src/pcapi/core/offers/api.py

```diff
@@ -145,16 +144,49 @@ def edit_stock(
         if bookings:
             try:
                 user_emails.send_batch_stock_postponement_emails_to_users(bookings, send_email=mailing.send_raw_email)
-            except mailing.MailServiceException as mail_service_exception:
+            except mailing.MailServiceException as exc:
                 # fmt: off
                 app.logger.exception(
-                    "Could not notify beneficiaries about update of stock=%s: exc",
+                    "Could not notify beneficiaries about update of stock=%s: %s",
                     stock.id,
-                    mail_service_exception,
+                    exc,
                 )
                 # fmt: on
 
     if feature_queries.is_active(FeatureToggle.SYNCHRONIZE_ALGOLIA):
         redis.add_offer_id(client=app.redis_client, offer_id=stock.offerId)
 
     return stock
+
+
+def delete_stock(stock: StockSQLEntity) -> None:
+    validation.check_stock_is_deletable(stock)
+
+    stock.isSoftDeleted = True
+
+    cancelled_bookings = []
+    for booking in stock.bookings:
+        if not booking.isCancelled and not booking.isUsed:
+            booking.isCancelled = True
+            cancelled_bookings.append(booking)
+
+    repository.save(stock, *cancelled_bookings)
+
+    if cancelled_bookings:
+        for callback, error_message in (
+            (
+                user_emails.send_batch_cancellation_emails_to_users,
+                "Could not notify beneficiaries about deletion of stock=%s: %s",
+            ),
+            (
+                user_emails.send_offerer_bookings_recap_email_after_offerer_cancellation,
+                "Could not notify offerer about deletion of stock=%s: %s",
+            ),
+        ):
+            try:
+                callback(cancelled_bookings, mailing.send_raw_email)
+            except mailing.MailServiceException as exc:
+                app.logger.exception(error_message, stock.id, exc)
```

Comment:
```suggestion
        try:
            user_emails.send_batch_cancellation_emails_to_users(cancelled_bookins, mailing.send_raw_email)
        except mailing.MailServiceException as exc:
            app.logger.exception("Could not notify beneficiaries about deletion of stock=%s: %s", stock.id, exc)
        try:
            user_emails.send_offerer_bookings_recap_email_after_offerer_cancellation(cancelled_bookins, mailing.send_raw_email)
        except mailing.MailServiceException as exc:
            app.logger.exception("Could not notify offerer about deletion of stock=%s: %s", stock.id, exc)
```

Je suis pas méga fan de cette syntaxe :thinking: 
Est-ce que les deux try except c'est pas finalement plus simple ? 
J'aime beaucoup l'idée de rajouter des infos dans les logs et d'être plus précis + on ne stoppe pas l'exécution si un des deux fail mais je trouve que la boucle for complexifie plus le code que ce qu'elle apporte en terme de DRY :)

Reply:
Tu as raison. À la relecture ma version est moins claire.